support/cas-server-support-cassandra-ticket-registry/src/main/java/org/apereo/cas/ticket/registry/CassandraTicketRegistry.java

```diff
@@ -0,0 +1,120 @@
+package org.apereo.cas.ticket.registry;
+
+import org.apereo.cas.cassandra.CassandraSessionFactory;
+import org.apereo.cas.ticket.BaseTicketSerializers;
+import org.apereo.cas.ticket.ExpirationPolicy;
+import org.apereo.cas.ticket.Ticket;
+import org.apereo.cas.ticket.TicketCatalog;
+import org.apereo.cas.ticket.TicketDefinition;
+
+import com.datastax.driver.mapping.Mapper;
+import com.datastax.driver.mapping.MappingManager;
+import com.datastax.driver.mapping.Result;
+import com.datastax.driver.mapping.annotations.Accessor;
+import com.datastax.driver.mapping.annotations.Query;
+import com.datastax.driver.mapping.annotations.QueryParameters;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * This is {@link CassandraTicketRegistry}
+ *
+ * @since 6.1.0
+ */
+public class CassandraTicketRegistry extends AbstractTicketRegistry {
```

Comment:
You will need to make sure this class is thoroughly tested. Examine other integration tests and add one for this registry as well. 

You will also need to ensure this registry option is documented, much like all others in CAS docs and TOCs, etc.